src/dfx/src/lib/extension/manager/execute.rs

```diff
@@ -1,19 +1,21 @@
 use super::ExtensionManager;
 use crate::lib::error::ExtensionError;
-use std::ffi::OsString;
+use std::path::Path;
 
 impl ExtensionManager {
     pub fn run_extension(
         &self,
-        extension_name: OsString,
-        params: Vec<OsString>,
+        dfx_cache: &Path,
+        extension_name: String,
+        mut params: Vec<String>,
     ) -> Result<(), ExtensionError> {
-        let extension_name = extension_name
-            .into_string()
-            .map_err(ExtensionError::InvalidExtensionName)?;
-
         let mut extension_binary = self.get_extension_binary(&extension_name)?;
 
+        params.extend([
+            "--dfx-cache-path".into(),
+            dfx_cache.to_string_lossy().into(),
```

Comment:
Killin' me with `to_string_lossy` on a path.  Is it really necessary?

Reply:
to me, it feels like a way to go because `Path` is backed by `OsString`. The only methods available to turn it into a string are:
- `display` https://doc.rust-lang.org/std/path/struct.Display.html
  > This may perform lossy conversion, depending on the platform
- `as_os_str` & `as_mut_os_str` - pointless here, just kicking the can down the road
- `to_str` which yields None in cases where `display` and `to_string_lossy` would perform lossy conversion
- [to_string_lossy](https://doc.rust-lang.org/std/path/struct.Path.html#method.to_string_lossy)
- `Debug` (`{:?}`), which seems to be the only way to perform a lossless conversion

If you insist on staying away from lossy conversion, I can revert https://github.com/dfinity/sdk/pull/3169/commits/1d5197f816c3ab7bcfeb7c7d900a9122cd648590 and we'll use OsStrings everywhere, which I think is also fine

Reply:
As mentioned above, `PathBuf` (not &Path) can convert directly to OsString with `PathBuf.into_os_string()`.  I leave it to you to decide if you'd rather pass PathBuf or OsString.

Reply:
but we're not using OsString anymore

params is now `Vec<String>`, therefore we need to convert from PathBuf to String

I honestly think sticking with `params: Vec<OsString>` will be better here

Reply:
There was no request to change params from Vec<OsString>, only the extension name, because it was then being converted to String.

Reply:
how about we turn all instances of extension_name to OsStr/OsString instead? https://sourcegraph.com/search?patternType=regexp&case=yes&q=context%3Aglobal+repo%3A%5Egithub%5C.com%2Fdfinity.%2B%24+extension_name&groupBy=repo

the name is never generated by the dfx, it always comes from the outside world (either passed as a string in CLI, or as a result of `listdir`)

Reply:
`String` seems like a better fit for `extension_name`.  Conceptually it's a command name, like `start` or `canister` or `nns`, even though we happen to be reading it from a list of directories.